src/sentry/static/sentry/app/views/settings/account/accountDetails.tsx

```diff
@@ -1,64 +1,53 @@
 import React from 'react';
 
-import {updateUser} from 'app/actionCreators/account';
+import {addErrorMessage, addSuccessMessage} from 'app/actionCreators/indicator';
 import AsyncView from 'app/views/asyncView';
 import AvatarChooser from 'app/components/avatarChooser';
 import Form from 'app/views/settings/components/forms/form';
 import JsonForm from 'app/views/settings/components/forms/jsonForm';
 import SettingsPageHeader from 'app/views/settings/components/settingsPageHeader';
 import accountDetailsFields from 'app/data/forms/accountDetails';
 import accountPreferencesFields from 'app/data/forms/accountPreferences';
+import {APIRequestMethod} from 'app/api';
+import {t} from 'app/locale';
 
 const ENDPOINT = '/users/me/';
 
 class AccountDetails extends AsyncView {
-  getEndpoints() {
+  getEndpoints(): ReturnType<AsyncView['getEndpoints']> {
     return [['user', ENDPOINT]];
   }
 
-  handleSubmitSuccess = user => {
-    updateUser(user);
-    this.setState({user});
```

Comment:
We should keep this even though it doesn't seem to do anything. `updateUser` updates our config store (problem is, no components listen to the ConfigStore, they just access it directly).

And the form itself does not update `this.state`, the form themselves keep their own state, but none of the components on this page use overlapping user properties. So maybe it's good to remove `setState`, but let's leave a comment in maybe `getEndpoints` that local state is NOT updated when the form saves.

Reply:
I was really expecting some feedback on this part of the code :) I was not sure about the `updateUser`, it didn't make any sense to me... I will add comments explaining the reason the `updateUser`  method should be there... and also the reason we don't update the state... thanks @billyvg 😉